src/networking.c

```diff
@@ -2717,7 +2717,10 @@ void readQueryFromClient(connection *conn) {
         atomicIncr(server.stat_net_input_bytes, nread);
     }
 
-    if (!(c->flags & CLIENT_MASTER) && sdslen(c->querybuf) > server.client_max_querybuf_len) {
+    if (!(c->flags & CLIENT_MASTER) &&
+        /* The commands cached in the MULTI/EXEC queue have not been executed yet,
+         * so they are also considered a part of the query buffer in a broader sense. */
+        (c->mstate.argv_len_sums + sdslen(c->querybuf) > server.client_max_querybuf_len)) {
```

Comment:
maybe we wanna take this opportunity and have a lower limit for unauthenticated clients?
like we do for multi-bulk length and bulk size?

Reply:
makes sense, what do you think would be an appropriate limit, like 1MB?

and unlike the limit on unauthenticated clients in multi-bulk, which replies with "Protocol error: unauthenticated multibulk length" before disconnecting, in this case, the connection is directly terminated and counted in the `client_query_buffer_limit_disconnections` metric. The documentation should also provide separate clarification on this.

Reply:
yes.. these "unauthenticated" connections can only run AUTH / HELLO commands. they have no reason to use MULTI anyway.
i think a limit of 32kb could be enough. but also ok with 1mb or alike.

Reply:
They could be doing a module authentication with a symmetric key or something, so I could see those being a bit longer, but I can't think of any reason a key would be more than 32kb (even AWS's IAM authentication signatures, which are very verbose, only tap out around 2ish kb). I guess I would rather do 1mb just to be very safe here, since this "could" be some obscure breaking change that would break hard.